libs/Functions.php

```diff
@@ -511,16 +511,24 @@ public function post_author($options=array()){
 	}
 
 	//Post type meta
-	public function post_type($icon=true){
+	public function post_type($options=array() ){
+		$defaults = apply_filters('nebula_post_type_defaults', array(
+			'icon' 		=> true, // boolean true for generic defaults, boolean false to disable icon, or string of class name(s) for icon.
+			'linked'	=> false // boolean true links output to the post type archive page
+		));
+		$data = array_merge($defaults, $options);
+
 		if ( get_theme_mod('search_result_post_types', true) ){
-			$post_icon_img = '<i class="fas fa-thumbtack"></i>';
-			if ( $icon ){
+			if( $data['icon'] ){
 				global $wp_post_types;
 				$post_type = get_post_type();
+				$post_type_labels = get_post_type_object( $post_type )->labels;
 
-				if ( $post_type === 'post' ){
+				if( gettype( $data['icon'] ) == 'string' && $data['icon'] != '' ){
+					$post_icon_img = '<i class="'.esc_html( $data['icon'] ).'"></i>';
+				}elseif ( $post_type === 'post' ){
 					$post_icon_img = '<i class="fas fa-fw fa-thumbtack"></i>';
-				} elseif ( $post_type === 'page' ){
+				} elseif ( $post_type === 'page' ){		
 					$post_icon_img = '<i class="fas fa-fw fa-file-alt"></i>';
 				} else {
```

Comment:
Technically this `else` is also unnecessary– I know that is carried over from the original writing, so I may re-visit that part myself later.

Reply:
Only you can conquer your past mistakes

Reply:
I refactored this while fixing another issue anyway